src/aws/osml/image_processing/map_tileset_factory.py

```diff
@@ -0,0 +1,41 @@
+#  Copyright 2024 Amazon.com, Inc. or its affiliates.
+
+from enum import Enum
+from typing import Optional
+
+from .map_tileset import MapTileSet
+from .map_tileset_wmq import WebMercatorQuadMapTileSet
+
+
+class WellKnownMapTileSet(str, Enum):
+    """
+    A partial list of well known tile sets used by this library.
+    """
+
+    WEB_MERCATOR_QUAD = "WebMercatorQuad"
+    WEB_MERCATOR_QUAD_X2 = "WebMercatorQuadx2"
+
+
+class MapTileSetFactory:
+    """
+    This class provides a means to construct / access the implementation of well known tile sets using their
+    name. It allows clients to easily work with the TileSet abstraction independent of any implementation details
+    associated with specific tile sets.
+    """
+
+    @staticmethod
+    def get_for_id(tile_matrix_set_id: str) -> Optional[MapTileSet]:
+        """
+        Constructs a tile set matching the requested id.
+
+        :param tile_matrix_set_id: the tile set id
+        :return: the TileSet or None if not available
+        """
+        if tile_matrix_set_id == WellKnownMapTileSet.WEB_MERCATOR_QUAD.value:
```

Comment:
Should the tile size me an initialization argument perhaps? Might that change?

Reply:
No; the tile size is specified by the tile set. In this specific instance the "WebMercatorQuad" tile set specifies 256x256 tiles and "WebMercatorQuadx2" is almost identical except each tile is 512x512. In the WebMercatorQuadMapTileSet implementation I made tile size an argument so the same code can work for either but here at the factory when an end user is asking for a specific tile set by ID then the tile size is implied.